cedar-policy-core/src/parser/cst_to_ast.rs

```diff
@@ -756,19 +760,21 @@ impl ExprOrSpecial<'_> {
                 }
             },
             Self::Var(var, _) => {
-                errs.push(err::ParseError::ToAST(format!(
+                errs.push(err::ParseError::InvalidString(format!(
                     "not a string literal: {var}"
                 )));
                 None
             }
             Self::Name(name) => {
-                errs.push(err::ParseError::ToAST(format!(
+                errs.push(err::ParseError::InvalidString(format!(
                     "not a string literal: {name}"
                 )));
```

Comment:
I wonder if the errors in this function are more `IllegalNonLiteral` errors that `InvalidString` errors? Also, looking at this code again, I think the `Var`, `Name`, `Expr` cases can be smooshed into one case. Same comments about the function below.

Reply:
Agree with `IllegalNonLiteral` part. I think it's reasonable to keep the separate errors for `Identifier` `Expr` and `String`. If we had more time it could be useful to try to explain why something isn't a valid `identifier` and if we do that we'd want to handle each case separately.